### PR TITLE
Add activation presets for NixOS and noop activation

### PR DESCRIPTION
**Pull request by balsoft**
It turns out that some activation scripts are very common, and I think having special functions for them would be nice.

### CODE REVIEW

**Comment by notgne2** [comment]
I feel like these should be namespaced

**Comment by balsoft** [comment]
Yeah, but then `setActivate` would look very strange unnamespaced. Move everything into `lib.activate.{noop,nixos,custom}` and `setActivate = builtins.trace "lib.setActivate is deprecated, use lib.activate.custom instead!" lib.activate.custom` ?

**Comment by notgne2** [comment]
sounds good to me

**Comment by notgne2** [comment]
Though I will add, I think the warning should actually refer to the existence of `noop` and `nixos` as options too, as like you said in the PR message, those are very common

File: flake.nix [hunk 1]
```diff
@@ -48,25 +48,32 @@
...
+                  executable = true;
+                  destination = "/deploy-rs-activate";
+                })
+              ];
+            };
 
-          noopActivate = base: setActivate base ":";
+            nixos = base: setActivate base "$PROFILE/bin/switch-to-configuration switch";
+
+            noop = base: setActivate base ":";
```
Comment (notgne2) [inline flake.nix:75]: doesn't this mean any usage of `activate.{nixos, noop}` is going to produce a deprecation warning?

**Pull request merged**